source/slang/slang-ir-simplify-cfg.cpp

```diff
@@ -765,7 +765,10 @@ static bool removeTrivialPhiParams(IRBlock* block)
         termInsts.add(termInst);
         for (UInt i = 0; i < termInst->getArgCount(); i++)
         {
-            if (args[i].areKnownValueSame)
+            // Self-referential parameters can be skipped, as they cannot
+            // introduce a new value. The phi can only have multiple different
+            // values if non-self-referential arguments differ.
+            if (args[i].areKnownValueSame && termInst->getArg(i) != params[i])
```

Comment:
If args[i].knownVal is nullptr, do we still want to assign the param to it as in line 774, even when the arg is self referential?

Reply:
AFAIK, there's no way to end up with a phi that only has self-referential arguments (that would implicate a block where all predecessors are back jumps), so `knownValue` will still always get some non-nullptr value in the end.

Reply:
Ah, that's true.